### PR TITLE
fix: add amount row for contract deployment

### LINKED ISSUE #27525: [Bug]: Contract Deployment - When I deploy a contract with ETH, I don't see the ETH amount I am sending in the Deployment screen
### Describe the bug

When I deploy a contract with ETH, I don't see the ETH amount I am sending in the Deployment screen. Notice, this happens when you have the `Estimate balance changes` feature disabled.
However, we should be able to see the amount we are sending with the deployment, as a standalone value (like we do with old screens), without depending on the Estimate balance changes feature.

### Expected behavior

We should see the amount we are sending with the deployment in the redesigned contract deployment screen, as we do in the old screen

### Screenshots/Recordings

https://github.com/user-attachments/assets/174f72a3-278c-4e23-99a2-4723b5f0dcda

### Steps to reproduce

1. Go to Remix
2. Trigger a contract deployment with some ETH
3. See no balance appears in the amount
4. Disable Transaction redesign
5. Trigger a contract deployment with some ETH
6. See balance appears in the amount

Example Contract

```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

contract Params {
    constructor() payable {}
     receive() external payable {
    }
}
```

### Error messages or log output

_No response_

### Detection stage

During release testing

### Version

12.4.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR adds an "Amount" row to display the token value for contract deployments when either:
- The transaction parameters include a value greater than 0, or
- The simulation fails.

This change ensures that users can see the token amount being sent during contract deployments, even when the "Estimate balance changes" feature is disabled.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27594?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27525

## **Manual testing steps**

1. Go to Remix
2. Trigger a contract deployment with some ETH
3. See no balance appears in the amount
4. Disable Transaction redesign
5. Trigger a contract deployment with some ETH
6. See balance appears in the amount

Example Contract

```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

contract Params {
    constructor() payable {}
     receive() external payable {
    }
}
```

### **Before**

[amount-contract.webm](https://github.com/user-attachments/assets/131f36f6-73f6-465e-b731-bb10554d5f76)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx [hunk 1]
```diff
@@ -83,6 +86,30 @@ export const MethodDataRow = () => {
   );
 };
 
+export const AmountRow = () => {
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx:89]: Do we need to export this if it's only used locally?

File: ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx [hunk 2]
```diff
@@ -83,6 +86,30 @@ export const MethodDataRow = () => {
...
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext<TransactionMeta>();
+  const { currency } = useUserPreferencedCurrency(PRIMARY);
+
+  const value = currentConfirmation?.txParams?.value;
+  const simulationData = currentConfirmation?.simulationData;
+
+  if (!value || value === '0x' || !simulationData?.error) {
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx:97]: Should the condition include `(simulationData && !simulationData.error)` to ensure this row doesn't flicker while simulation is loading and only shows after simulation fails?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

**Comment by sleepytanya** [comment]
Amount is displayed when the "Estimate balance changes" is disabled:


https://github.com/user-attachments/assets/af004974-e4d4-414c-8f9b-1c7c39a948ea


estimationOn


estimationOff

File: ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx (hunk 1 above)
Comment (vinistevam) [reply ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.tsx]: Done https://github.com/MetaMask/metamask-extension/pull/27594/commits/b2ff77a324f3e386cf78b8b99f8bae25435c91f5

File: test/data/confirmations/contract-interaction.ts [hunk 3]
```diff
@@ -160,4 +164,13 @@ export const genUnapprovedContractInteractionConfirmation = ({
     userEditedGasLimit: false,
     userFeeLevel: 'medium',
     verifiedOnBlockchain: false,
-  } as SignatureRequestType);
+  } as SignatureRequestType;
+
+  // Overwrite simulation data if provided
+  if (simulationData) {
+    console.log('>>>>>', simulationData);
```
Comment (OGPoyraz) [inline test/data/confirmations/contract-interaction.ts:171]: I guess this is forgotten

**Review by OGPoyraz — DISMISSED** [review]

File: test/data/confirmations/contract-interaction.ts (hunk 3 above)
Comment (vinistevam) [reply test/data/confirmations/contract-interaction.ts]: removed https://github.com/MetaMask/metamask-extension/pull/27594/commits/1aec7d843977cf3324937f1377891a499af9fff0

**Review by pedronfigueiredo — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Pull request merged**